include/nil/blueprint/component_manifest_utilities.hpp

```diff
@@ -0,0 +1,73 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2023 Alexey Kokoshnikov <alexeikokoshnikov@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#ifndef CRYPTO3_ASSIGNER_COMPONENT_MANIFEST_UTILITIES_HPP
+#define CRYPTO3_ASSIGNER_COMPONENT_MANIFEST_UTILITIES_HPP
+
+#include <vector>
+#include <array>
+
+#include <nil/blueprint/manifest.hpp>
+
+namespace nil {
+    namespace blueprint {
+
+        static const std::array<uint32_t, 0> empty_array = {};
```

Comment:
Since these two are not part of the public interface - maybe we shall move them into nil::blueprint::detail namespace?

Reply:
Yes, but actually if make <COMPONENT>::component_type public member - we'll no need these 2 ugly constants. We have talk about that with Dmitry Tabalin.